### ISSUE TITLE
NullPointerException in IgnoredReturnValue rule in version 1.23.0.

### ISSUE THREAD

**Issue by MaximSysoev** [detekt/detekt#6175]
## Expected Behavior
IgnoredReturnValue rule should check successfully.

## Observed Behavior
The rule crashes during `./gradlew detekt` with `java.lang.NullPointerException` when it is trying find package because containingDeclaration in DeclarationDescriptor is null.
```
Caused by: java.lang.NullPointerException
        at org.jetbrains.kotlin.backend.common.serialization.LegacyDescriptorUtilsKt.findPackage(LegacyDescriptorUtils.kt:29)
        at io.gitlab.arturbosch.detekt.rules.bugs.IgnoredReturnValue.visitCallExpression(IgnoredReturnValue.kt:118)
        at org.jetbrains.kotlin.psi.KtVisitorVoid.visitCallExpression(KtVisitorVoid.java:811)
        at org.jetbrains.kotlin.psi.KtVisitorVoid.visitCallExpression(KtVisitorVoid.java:21)
        ...
```

## Steps to Reproduce
- update detekt to 1.23.0.
- run ./gradlew detekt

## Context
Before updating of the version this rule was checking correctly. Then I update to 1.23.0 and this rule is crashing now. 
So it seems, that the problem in method which get local contacts via Cursor inside `kotlinx.coroutines.flow.callbackFlow `
When I comment few strings as in example, it is checking correctly.
```
callbackFlow {
    // getting cursore

//    val contentObserver =
//      object : ContentObserver(createHandler()) {
//
//        override fun onChange(selfChange: Boolean) {
//          launch {
//            if (cursor != null) {
//              send(cursor)
//            }
//          }
//        }
//      }
//
//    cursor?.registerContentObserver(contentObserver)
//
//    awaitClose {
//      cursor?.unregisterContentObserver(contentObserver)
//      cursor?.close()
//    }
  }
```

## Your Environment
* Version of detekt used: 1.23.0
* Version of Gradle used (if applicable): 8.0.1
* Operating System and version: Mac OS X 13.0 (aarch64)

**Comment by atulgpt** [comment]
Hi, @maxsdev the above rule only run when type resolution is enabled. But in the steps to repro you gave `./gradlew detekt` command which runs without type resolution(hence `IgnoredReturnValue` will be skipped). Also, can you clarify `When I comment few strings as in example, it is checking correctly.` is provided example running correctly by the rule

**Comment by diastremskii** [comment]
Hey! 👋 
I've encountered this issue as well, and for me it reproduces with a certain combination of dependencies 🤔 
I've quickly drafted this repo as an example: https://github.com/diastremskii/detekt-issue-6175. In this repo, `./gradlew detektMain` fails with
```
> Analyzing /Users/user/Documents/Kotlin/detekt-issue-6175/detekt-issue-6175/src/main/kotlin/Main.kt led to an exception.
  Location: org.jetbrains.kotlin.backend.common.serialization.LegacyDescriptorUtilsKt.findPackage(LegacyDescriptorUtils.kt:29)
  The original exception message was: null
```
Certainly, there must be a way to reproduce it without pulling in SLF4J and Arrow-kt and I'm sorry for building not very clean example, but at least it captures and reproduces the issue.

**Comment by BraisGabin** [comment]
I'm able to reproduce the issue. And I have no idea why but if I update the gradle version to the last one it works again: `./gradlew wrapper --gradle-version 8.2.1`

Could you check if that happens to you too?

**Comment by diastremskii** [comment]
Interesting 🤔 
Yes, I can confirm, it doesn't reproduce anymore on Gradle 8.2.1

**Comment by maxsdev** [comment]
> Hi, @maxsdev the above rule only run when type resolution is enabled. But in the steps to repro you gave `./gradlew detekt` command which runs without type resolution(hence `IgnoredReturnValue` will be skipped). Also, can you clarify `When I comment few strings as in example, it is checking correctly.` is provided example running correctly by the rule.
> 
> If you can provide a failing TC that would be great

Hi, sorry for long time answer. Yes, provided example with commented code is running correctly.
I suppose that if `IgnoredReturnValue` should be skipped, that it shouldn't crash. Or have I missed something?

**Comment by BraisGabin** [comment]
Can you update the gradle version? It seems that it fixes the issue.

**Comment by maxsdev** [comment]
> Can you update the gradle version? It seems that it fixes the issue. 

Ok, I'll try

**Comment by AlbRoehm** [comment]
Same issue occured for me after updating kotlin, gradle and detekt plugin today (kotlin: 1.9.10, gradlew: 8.3, detekt plugin: 1.23.1). I could narrow it down to a single line of code which is creating the issue `fileContent.readBytes()` where readBytes() is the stdlib function from `kotlin.io.ByteStreams`. If this line is replaced by `byteArrayOf()` for example detekt is happy.

```
Execution failed for task ':detektMain'.
> Analyzing /Users/.../SettlementClient.kt led to an exception.
  Location: org.jetbrains.kotlin.backend.common.serialization.LegacyDescriptorUtilsKt.findPackage(LegacyDescriptorUtils.kt:29)
  The original exception message was: null
  Running detekt '1.23.1' on Java '17.0.8.1+1' on OS 'Mac OS X'
  If the exception message does not help, please feel free to create an issue on our GitHub page.
```

**Comment by BraisGabin** [comment]
I researched a bit more on this issue. I used the @diastremskii sample and check the difference between how we execute gradle 7 and gradle 8 executes detekt. And I saw that with gradle 8 the detekt gradle plugin is passing this parameters: 
```
--jvm-target 17
--jdk-home /Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home
```
And gradle 7 doesn't. So I assume that the problem is related with that.

I got this information adding the property `detekt-dry-run=true` in the gradle.properties file.

I added this to the configuration:

```
tasks.withType<io.gitlab.arturbosch.detekt.Detekt>().configureEach {
    this.jvmTarget = "17"
}
```

And the error persists but now the only difference between gradle versions is the `--jdk-home`

I'm not 100% how to continue from here. Any ideas @3flex?

---

Meanwhile, the people that can reproduce this issue, can you add `detekt-dry-run=true` to `gradle.properties` and rerun detekt? you should get a list with the arguments that the detekt gradle plugin passes to the detekt cli. Is in that list the `jdk-home`? Which version of gradle are you using? Do you have `JAVA_HOME` defined on your environment?

**Comment by AlbRoehm** [comment]
Hey, sorry i got a bit disconnected from this issue since some other things came up and i did not see it anymore. Honestly i also cannot remember what it fixed it back then. 
I came back to this because today when i started seeing these errors again after i added a context receiver to a function. Also i cannot really find a clear rule or pattern what breaks it consistently. E.g. adding ktor.Application as context-receivers works fine. But adding my own class breaks detekt. But adding only part of the class with an interface is fine though?? Sorry, that i'm not able to provide a better picture at the moment. Maybe this still rings a bell somewhere.

About the dry run, the --jdk-home & the --java-target is present and set correctly according to gradle settings.  I checked these to version of jdk (java_home is set to tem-17.)
```
--jvm-target 21 --jdk-home /Users/xyz.sdkman/candidates/java/21.0.2-tem 
--jvm-target 17 --jdk-home /Users/xyz/.sdkman/candidates/java/17.0.10-tem
```

```
Using:
kotlin 1.9.22
gradle-wrapper 8.5
detekt 1.23.4
```

**Comment by 3flex** [comment]
I looked at this, there are a few things in play.

For the reproducer provided it works on Gradle 8 because toolchains are used unconditionally: https://docs.gradle.org/current/userguide/upgrading_version_7.html#toolchain_based_tasks_for_jvm_projects, but on Gradle 7.6.x toolchain has to be setup manually. When this isn't done, Java types cannot be resolved, which isn't handled properly in the rule. And because the type's not resolved, the `!!` used in the `findPackage()` function causes a NPE.

Running detekt with debug enabled shows:
```
Main.kt:11:58: error: none of the following functions can be called with the arguments supplied: 
public open fun getLogger(p0: Class!): Logger! defined in org.slf4j.LoggerFactory
public open fun getLogger(p0: String!): Logger! defined in org.slf4j.LoggerFactory
            .tapLeft { error: Throwable -> LoggerFactory.getLogger("name").isErrorEnabled() }
                                                         ^
error: supertypes of the following classes cannot be resolved. Please make sure you have the required dependencies in the classpath:
    class org.slf4j.LoggerFactory, unresolved supertypes: java.lang.Object
```

Options:
1. Fix outstanding issues with type resolution, enable by default, and stop hiding the compiler errors behind the debug flag. This is my preferred option.
2. Promote use of toolchains on Gradle < 8. Perhaps a warning when running analysis task when no JDK home is set.
3. Update rule logic to defend against this case occurring. Will lead to false negatives.

@AlbRoehm when you run detekt with debug enabled, do you see any compiler errors similar to the one above? Which types aren't being resolved?

**Comment by 3flex** [comment]
Sorry for misleading you, you need to enable `debug` for detekt which doesn't use that flag (but probably should...)

```kotlin
detekt {
    debug = true
}
```

**Comment by AlbRoehm** [comment]
This produced quite some more output :)  Sidenote, contex-receivers are enabled
```
    withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile> {
        kotlinOptions {
            freeCompilerArgs = freeCompilerArgs + "-Xcontext-receivers"
        }
    }
```


```
> Task :detektMain FAILED
Loaded extensions:

Registered rule sets:
io.gitlab.arturbosch.detekt.rules.bugs.PotentialBugProvider@5acbca82
io.gitlab.arturbosch.detekt.rules.complexity.ComplexityProvider@4ba6fa30
io.gitlab.arturbosch.detekt.rules.coroutines.CoroutinesProvider@7d11aae2
io.gitlab.arturbosch.detekt.rules.documentation.CommentSmellProvider@284de925
io.gitlab.arturbosch.detekt.rules.empty.EmptyCodeProvider@2c56a974
io.gitlab.arturbosch.detekt.rules.exceptions.ExceptionsProvider@505db1b4
io.gitlab.arturbosch.detekt.rules.naming.NamingProvider@206b57e7
io.gitlab.arturbosch.detekt.rules.performance.PerformanceProvider@6728860e
io.gitlab.arturbosch.detekt.rules.style.StyleGuideProvider@7e2c8e96
warning: classpath entry points to a non-existent location: /Users/albrecht/workspace/build/classes/java/main
//Users/albrecht/workspace/src/main/kotlin/demo/backend/Application.kt:32:9: error: not enough information to infer type variable Error
either {
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/Application.kt:35:35: error: to use contextual declarations, specify the `-Xcontext-receivers` compiler option
val (env, deps) = initDependencies().bind()
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/Application.kt:43:20: error: cannot infer a type for this parameter. Please specify it explicitly.
}.onLeft { error -> logger.error(error.toString()) }
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/Application.kt:60:5: error: to use contextual declarations, specify the `-Xcontext-receivers` compiler option
installFeatures(deps)
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/Application.kt:63:13: error: to use contextual declarations, specify the `-Xcontext-receivers` compiler option
installUserProfile(deps)
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/database/ExposedDB.kt:101:1: error: the feature "context receivers" is experimental and should be enabled explicitly
context(ResourceScope)
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/database/ExposedDB.kt:103:5: error: type mismatch: inferred type is Resource<ExposedDB> but ExposedDB was expected
autoCloseable {
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/database/ExposedDB.kt:103:5: error: type mismatch: inferred type is Resource<ExposedDB> but ExposedDB was expected
autoCloseable {
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/database/ExposedDB.kt:107:1: error: the feature "context receivers" is experimental and should be enabled explicitly
context(ResourceScope)
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/database/ExposedDB.kt:109:5: error: type mismatch: inferred type is Resource<HikariDataSource> but DataSource was expected
autoCloseable {
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/database/ExposedDB.kt:109:5: error: type mismatch: inferred type is Resource<HikariDataSource> but DataSource was expected
autoCloseable {
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/env/Dependencies.kt:24:1: error: the feature "context receivers" is experimental and should be enabled explicitly
context(ResourceScope)
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/env/Dependencies.kt:26:5: error: not enough information to infer type variable Error
either {
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/env/Dependencies.kt:27:19: error: to use contextual declarations, specify the `-Xcontext-receivers` compiler option
val env = env()
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/env/Dependencies.kt:28:28: error: to use contextual declarations, specify the `-Xcontext-receivers` compiler option
val dependencies = dependencies(env)
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/env/Dependencies.kt:28:41: error: type mismatch: inferred type is Any but Env was expected
val dependencies = dependencies(env)
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/env/Dependencies.kt:33:1: error: the feature "context receivers" is experimental and should be enabled explicitly
context(ResourceScope, Raise<Problem>)
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/env/Dependencies.kt:35:22: error: to use contextual declarations, specify the `-Xcontext-receivers` compiler option
val dataSource = dataSource(env.dataSource)
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/env/Dependencies.kt:36:31: error: to use contextual declarations, specify the `-Xcontext-receivers` compiler option
val database: IDatabase = exposedDB(dataSource)
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/env/Dependencies.kt:43:37: error: type mismatch: inferred type is Resource<HgClient> but IHgClient was expected
val hgService = hgService(hgClient)
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/env/Dependencies.kt:45:39: error: unresolved reference: raise
runMigration(dataSource).onLeft { raise(it) }
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/env/Env.kt:29:1: error: the feature "context receivers" is experimental and should be enabled explicitly
classpath entry points to a non-existent location: /Users/albrecht/workspace/build/classes/java/main

context (Raise<Problem>)
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/env/Env.kt:35:9: error: unresolved reference: raise
raise(InvalidEnvironmentError("Environment variables are misconfigured", it))
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/migration/flyway.kt:15:39: error: to use contextual declarations, specify the `-Xcontext-receivers` compiler option
is FlywayException -> createMigrationBaseline(dataSource)
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/migration/flyway.kt:21:1: error: the feature "context receivers" is experimental and should be enabled explicitly
context (Raise<DatabaseError>)
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/migration/flyway.kt:23:50: error: type mismatch: inferred type is Unit but MigrateResult was expected
catch({ migrateDatabase(dataSource, true) }) {
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/migration/flyway.kt:24:9: error: unresolved reference: raise
raise(DatabaseError(it.message ?: "Migration Error", it))
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/Features.kt:22:1: error: the feature "context receivers" is experimental and should be enabled explicitly
context(Application)
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/Features.kt:24:5: error: unresolved reference: routing
routing {
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/Features.kt:25:9: error: unresolved reference: authenticate
authenticate(AUTH_NAME) {
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/Features.kt:26:13: error: unresolved reference: post
post("purchase/features/{featureId}") {
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/Features.kt:28:36: error: unresolved reference. None of the following candidates is applicable because of receiver type mismatch: 
public val PipelineContext<*, ApplicationCall>.call: ApplicationCall defined in io.ktor.server.application
val userName = call.principal<UserIdPrincipal>()?.name.orEmpty()
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/Features.kt:31:25: error: unresolved reference. None of the following candidates is applicable because of receiver type mismatch: 
public val PipelineContext<*, ApplicationCall>.call: ApplicationCall defined in io.ktor.server.application
call.parameters["featureId"]?.toIntOrNull() == 1
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/Features.kt:33:28: error: unresolved reference. None of the following candidates is applicable because of receiver type mismatch: 
public val PipelineContext<*, ApplicationCall>.call: ApplicationCall defined in io.ktor.server.application
ensure(call.receive<FeatureRequest>().encryptedToken.isNotBlank()) { "Encrypted token is blank" }
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/Features.kt:34:39: error: suspend function 'log' should be called only from a coroutine or another suspend function
deps.auditService.log(LogType.FeaturePurchase, "Feature purchase request for user $userName").bind()
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/Features.kt:36:49: error: suspend function 'read' should be called only from a coroutine or another suspend function
val user = deps.userService.read(userName).bind().toEither { InvalidInputError("User not found") }.bind()
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/Features.kt:44:56: error: suspend function 'update' should be called only from a coroutine or another suspend function
val updatedUser = deps.userService.update(user.copy(features = user.features.toSet().plus(1).toList())).bind()
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/Features.kt:46:21: error: unresolved reference. None of the following candidates is applicable because of receiver type mismatch: 
public val PipelineContext<*, ApplicationCall>.call: ApplicationCall defined in io.ktor.server.application
call.respond(HttpStatusCode.OK, updatedUser)
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/Features.kt:47:28: error: cannot infer a type for this parameter. Please specify it explicitly.
}.onLeft { left ->
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/Features.kt:48:39: error: suspend function 'log' should be called only from a coroutine or another suspend function
deps.auditService.log(LogType.WalletRegistration, left.toJson().getOrNull().orEmpty())
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/Features.kt:49:21: error: unresolved reference: call
call.respond(HttpStatusCode.BadRequest, left)
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/UserProfile.kt:24:1: error: the feature "context receivers" is experimental and should be enabled explicitly
context(Application, IUserService, IAuditService)
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/UserProfile.kt:27:5: error: unresolved reference: routing
routing {
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/UserProfile.kt:28:22: error: type mismatch: inferred type is String but Boolean was expected
authenticate(AUTH_NAME) {
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/UserProfile.kt:28:22: error: no value passed for parameter 'p1'
authenticate(AUTH_NAME) {
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/UserProfile.kt:28:22: error: no value passed for parameter 'p2'
authenticate(AUTH_NAME) {
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/UserProfile.kt:28:22: error: no value passed for parameter 'p3'
authenticate(AUTH_NAME) {
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/UserProfile.kt:28:22: error: no value passed for parameter 'p4'
authenticate(AUTH_NAME) {
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/UserProfile.kt:28:22: error: no value passed for parameter 'p5'
authenticate(AUTH_NAME) {
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/UserProfile.kt:28:22: error: no value passed for parameter 'p6'
authenticate(AUTH_NAME) {
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/UserProfile.kt:28:22: error: no value passed for parameter 'p7'
authenticate(AUTH_NAME) {
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/UserProfile.kt:28:22: error: no value passed for parameter 'p8'
authenticate(AUTH_NAME) {
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/UserProfile.kt:28:33: error: type mismatch: inferred type is () -> Unit but Boolean was expected
authenticate(AUTH_NAME) {
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/UserProfile.kt:29:13: error: unresolved reference: get
get("/user/profile") {
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/UserProfile.kt:31:36: error: unresolved reference. None of the following candidates is applicable because of receiver type mismatch: 
public val PipelineContext<*, ApplicationCall>.call: ApplicationCall defined in io.ktor.server.application
val userName = call.principal<UserIdPrincipal>()?.name.orEmpty()
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/UserProfile.kt:34:21: error: unresolved reference: log
log(LogType.User, "User profile request for $userName")
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/UserProfile.kt:36:32: error: unresolved reference: read
val user = read(userName).bind()
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/UserProfile.kt:40:21: error: unresolved reference. None of the following candidates is applicable because of receiver type mismatch: 
public val PipelineContext<*, ApplicationCall>.call: ApplicationCall defined in io.ktor.server.application
call.respond(HttpStatusCode.OK, user)
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/UserProfile.kt:41:28: error: cannot infer a type for this parameter. Please specify it explicitly.
}.onLeft { left ->
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/UserProfile.kt:42:35: error: unresolved reference: log
val nextLog = log(LogType.WalletRegistration, left.toJson().getOrNull().orEmpty())
^
//Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/UserProfile.kt:43:21: error: unresolved reference: call
call.respond(HttpStatusCode.BadRequest, left)
^
Loaded extensions:
io.gitlab.arturbosch.detekt.rules.documentation.LicenceHeaderLoaderExtension@452fe68
io.github.detekt.metrics.processors.ProjectSLOCProcessor@4849dd96
io.github.detekt.metrics.processors.ProjectLOCProcessor@255d001d
io.github.detekt.metrics.processors.ProjectCLOCProcessor@5aebee89
io.github.detekt.metrics.processors.ProjectLLOCProcessor@4959fa02
io.github.detekt.metrics.processors.ProjectCognitiveComplexityProcessor@4ffa3bd2
io.github.detekt.metrics.processors.ProjectComplexityProcessor@2ea87d8
io.github.detekt.metrics.processors.PropertyCountProcessor@770beaa
io.github.detekt.metrics.processors.FunctionCountProcessor@17fddac
io.github.detekt.metrics.processors.ClassCountProcessor@1fc542fa
io.github.detekt.metrics.processors.PackageCountProcessor@4947d29d
io.github.detekt.metrics.processors.KtFileCountProcessor@4b40b0f1
Registered rule sets:
io.gitlab.arturbosch.detekt.rules.bugs.PotentialBugProvider@47a62b96
io.gitlab.arturbosch.detekt.rules.complexity.ComplexityProvider@66ad5302
io.gitlab.arturbosch.detekt.rules.coroutines.CoroutinesProvider@2bddaef9
io.gitlab.arturbosch.detekt.rules.documentation.CommentSmellProvider@44c9b8c3
io.gitlab.arturbosch.detekt.rules.empty.EmptyCodeProvider@3245d157
io.gitlab.arturbosch.detekt.rules.exceptions.ExceptionsProvider@3e7ca15e
io.gitlab.arturbosch.detekt.rules.naming.NamingProvider@78906a0e
io.gitlab.arturbosch.detekt.rules.performance.PerformanceProvider@7196ca7f
io.gitlab.arturbosch.detekt.rules.style.StyleGuideProvider@6d456273
Phase LoadConfig took 1ms
Phase CreateSettings took 1ms
Phase ValidateConfig took 3ms
Phase Parsing took 24ms
Phase Binding took 599ms
Phase LoadingExtensions took 1ms

Execution failed for task ':detektMain'.
> Analyzing /Users/albrecht/workspace/src/main/kotlin/demo/backend/routes/UserProfile.kt led to an exception.
Location: org.jetbrains.kotlin.backend.common.serialization.LegacyDescriptorUtilsKt.findPackage(LegacyDescriptorUtils.kt:29)
The original exception message was: null
Running detekt '1.23.4' on Java '17.0.10+7' on OS 'Mac OS X'
If the exception message does not help, please feel free to create an issue on our GitHub page.
```

One thing that leaves me quite puzzled. when i add a value assignment to the log functions that seems to cause the issue, the issue is not happening anymore and also debug error/warnings are not present anymore. 

```kotlin
before:
log(LogType.User, "User profile request for $userName")
-
val unused = log(LogType.User, "User profile request for $userName")
```

**Comment by 3flex** [comment]
Some of those issues are likely due to the classpath not being set correctly for the detektMain task - it leads to this problem but has a different root cause. Without knowing more about your project & dependencies it's hard to say for sure. But I believe in your case there are different causes unrelated to the original issue here. You might start a discussion to ask for support for configuring your project correctly.

> when i add a value assignment to the log functions that seems to cause the issue, the issue is not happening anymore.

Not sure why without investigating further.

> `error: to use contextual declarations, specify the `-Xcontext-receivers` compiler option`

To fix this we need #4465. We probably need that resolved for detekt 2, so I'll look at that again soon.

**Comment by AlbRoehm** [comment]
Adding a /src/main/java/EmptyClass.java class fixes this warning but leaves the rest of the issue unchanged. I see the same after just creating a project with the ktor-project-wizard in intellij. Sadly i'm still not able to create a minimal example. Was only able to get similar outputs about context-receiver but not an error. 
Thanks anyway for taking the time and looking into it.
